### DISCUSSION TITLE
Opencv 4.4 does not use Accelerate

### DISCUSSION THREAD

**Discussion by nosleduc** [Homebrew/discussions#2685] [Everyday usage]
Why opencv 4.5.4 is using openblas instead of Accelerate ? Accelerate is a better library for blas and lapack

**Comment by SMillerDev** [comment]
Because it was already used anyway: https://github.com/Homebrew/homebrew-core/pull/51182 you can make a pull request if you think it shouldn't be.

**Comment by carlocab** [comment]
Accelerate is old and buggy, and BLAS implementations shouldn't be mixed. Other formulae in OpenCV's dep tree use OpenBLAS, so it should use OpenBLAS too.

**Reply by nosleduc** [reply]
Old does not mean useless or inefficient.

I'm using Accelerate in my 3d reconstruction software with suite-sparse, ceres and opencv for more than 6 years and I did not notice bug. Until now, I'm compiling all my dependencies by hand and I would like to know if was easier to use homebrew instead.

Accelerate framework allows to use an optimized version of blas/lapack based on hardware components without worrying about the architecture (Intel or ARM).

It seems that on M1 processors, Accelerate is better than OpenBLAS
[Benchmark on m1](https://github.com/danielchalef/openblas-benchmark-m1)

**Reply by SMillerDev** [reply]
> It seems that on M1 processors, Accelerate is better than OpenBLAS

Only on small matrices, and as mentioned by Carlocab: it's buggy. You might not have encountered bugs, but other people have reported them to Homebrew and Apple, convincing us to generally avoid the framework.

If you have different requirements I would suggest making your own builds of dependencies so you can tweak them to your liking.

**Reply by carlocab** [reply]
See also: https://github.com/scipy/scipy/wiki/Dropping-support-for-Accelerate
